machine/v1/types_alibabaprovider.go

```diff
@@ -111,7 +123,9 @@ type AlibabaCloudMachineProviderConfig struct {
 	// +optional
 	SystemDisk SystemDiskProperties `json:"systemDisk,omitempty"`
 
-	// VSwitch is a reference to the vswitch to use for this instance
+	// VSwitch is a reference to the vswitch to use for this instance.
+	// A reference holds either the vSwitch ID, the resource name, or the required tags to search.
+	// When more than one vSwitch is returned for a tag search, only the first vSwitch returned will be used.
```

Comment:
is this ordering guaranteed to be consistent from the API you're searching with?

Reply:
that is a great question, it's not immediately clear from the docs[0] if the ordering is consistent. i will have to confer with alibaba about this. should we add a note here about that?


[0] https://www.alibabacloud.com/help/en/doc-detail/110425.htm?spm=a2c63.p38356.0.0.322619b8Xb6y0V#doc-api-Ecs-ListTagResources